### DISCUSSION TITLE
Standalone apps - php.ini configuration

### DISCUSSION THREAD

**Discussion by bpolaszek** [php/frankenphp#510] [Q&A]
Hi again, sorry for flooding the discussions with all my questions 😓 

After packing a standalone app, I created a dummy php file doing this:
```php
<?php
// memory-limit.php

var_dump(ini_get('memory_limit'));
```

Running `./my-frankenphp-app php-cli memory-limit.php` reported the usual `128M` value to me, which is likely to be insufficient in most apps. 

Is there a way to feed the build process with a `php.ini` (or even better, to make the standalone app get it at runtime)?

**Comment by withinboredom** [comment]
> which is likely to be insufficient in most apps.

I would be really shocked if apps are using more than that. That's a pretty sane production number. I've worked on PHP with billions of users accessing our site and our front-end servers had that amount. Our queue consumers had 256mb of memory. I even had to write jobs that iterated over millions of users at a time and it fit comfortably in 256mb of memory. This was in PHP 7.1 days.

If you need more than that, you probably have a memory leak or are doing something very strange; at least in my experience.

**Reply by bpolaszek** [reply]
I actually never hit the limit in HTTP context, but the apps I usually work on involve lots of data processing (in CLI context, this is why I'd like to use the standalone php-cli)

**Reply by bpolaszek** [reply]
Anyway, that doesn't answer my question.
What I'm thinking of at first is tweaking the memory limit, but it could be plenty of other PHP settings.

**Reply by nickchomey** [reply]
@withinboredom you haven't worked with a dynamic WordPress site then... Very common to use well beyond 256mb.

Anyway, to OP's point, it could be any php config variable, not just memory

**Reply by withinboredom** [reply]
> you haven't worked with a dynamic WordPress site then

@nickchomey -- this was at Automattic for WordPress.com... arguably one of the most dynamic and largest multi-site installs of WordPress in existence.

> What I'm thinking of at first is tweaking the memory limit, but it could be plenty of other PHP settings.

👍 I answered on mobile and thought this was on the already closed issue for this feature instead of a totally new one! This has already been merged btw: #497, nearly a week ago, so I was just debating the premise.

> the apps I usually work on involve lots of data processing

I highly recommend handling things in a streaming-kind-of-way (a healthy use of generators), so you don't use a bunch of memory. I also recommend not using `fgetcsv` since there has been a memory leak in it for over a decade (though, technically it isn't leaking, so it's "not a bug", but if you use any bit of a string from it, the entire string will stay in memory forever). In a WordPress context (to @nickchomey's point), you should also make sure you clear out the database's query log every so often and run `gc_collect_cycles()` every so often as well. But you should be able to process hundreds of gigabytes of data and stay well below 128mb of memory, you just have to make sure to clean up after yourself as you go.

**Reply by nickchomey** [reply]
I'm talking about a WordPress site that runs a combination of heavy, largely uncachable plugins, like Buddyboss + learndash + gamipress + woocommerce and much more. I can't imagine many (or even any) sites like that are running on wp.com. 

Again, they often use over 256mb and I've heard reports of using over 512 on sites that have lots of active users making hundreds of comments on each buddyboss/press activity feed post.

Of course, it doesn't NEED to use that much memory - admins can configure it so that plugins load selectively per-page, and the plugins themselves should be more efficient (eg don't dynamically load 3000 comments that no one is going to read - instead load 5 and add a read more Ajax button). Wordpress is a mess... 

Anyway, yeah I was glad to see the custom ini path feature added recently! Thanks for all your hard work!
